src/components/common/user-avatar/user-avatar.tsx

```diff
@@ -1,24 +1,56 @@
 /*
- * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
  *
  * SPDX-License-Identifier: AGPL-3.0-only
  */
 
 import React, { useMemo } from 'react'
 import { useTranslation } from 'react-i18next'
 import { ShowIf } from '../show-if/show-if'
-import Image from 'next/image'
 import styles from './user-avatar.module.scss'
+import type { UserInfo } from '../../../api/users/types'
+import { customizeAssetsUrl } from '../../../utils/customize-assets-url'
+import { getUser } from '../../../api/users'
+import { useAsync } from 'react-use'
 
-export interface UserAvatarProps {
+interface UserAvatarPropsUser {
+  user: UserInfo | null
+  username?: undefined
+}
+
+interface UserAvatarPropsUsername {
+  user?: undefined
+  username: string | null
+}
+
+interface UserAvatarPropsCommon {
   size?: 'sm' | 'lg'
-  name: string
-  photo: string
   additionalClasses?: string
   showName?: boolean
 }
 
-const UserAvatar: React.FC<UserAvatarProps> = ({ name, photo, size, additionalClasses = '', showName = true }) => {
+export type UserAvatarProps = UserAvatarPropsCommon & (UserAvatarPropsUser | UserAvatarPropsUsername)
+
```

Comment:
This type configuration still allows that I provide BOTH, but username will override user in L68-70.
I guess what you wanted to achieve is that I MUST provide at least one of these, right?
A quick look at the uses of this component shows that you always use only one of the props but never both.
You could just make "user" a combined type with "UserInfo | string" but the correct react way would be to create an intermediate component that takes a username, retrieves the user information (like you do in this component) and then forward this object to a component that renders it.

Or: Split the responsibilities into more components.


![image](https://user-images.githubusercontent.com/6124140/161420258-37f1ce01-e48f-4448-b8ba-3fabc93199cd.png)